requestmanager/asyncloader/asyncloader.go

```diff
@@ -76,23 +73,49 @@ func (al *AsyncLoader) Shutdown() {
 	al.cancel()
 }
 
+// RegisterPersistenceOption registers a new loader/storer option for processing requests
+func (al *AsyncLoader) RegisterPersistenceOption(name string, loader ipld.Loader, storer ipld.Storer) error {
+	if name == "" {
+		return errors.New("Persistence option must have a name")
+	}
+	response := make(chan error, 1)
+	select {
+	case <-al.ctx.Done():
+		return errors.New("context closed")
+	case al.incomingMessages <- &registerPersistenceOptionMessage{name, loader, storer, response}:
+	}
+	select {
+	case <-al.ctx.Done():
+		return errors.New("context closed")
+	case err := <-response:
+		return err
+	}
+}
+
 // StartRequest indicates the given request has started and the manager should
 // continually attempt to load links for this request as new responses come in
-func (al *AsyncLoader) StartRequest(requestID graphsync.RequestID) {
+func (al *AsyncLoader) StartRequest(requestID graphsync.RequestID, persistenceOption string) error {
+	response := make(chan error, 1)
+	select {
+	case <-al.ctx.Done():
+		return errors.New("context closed")
+	case al.incomingMessages <- &startRequestMessage{requestID, persistenceOption, response}:
+	}
 	select {
 	case <-al.ctx.Done():
-	case al.incomingMessages <- &startRequestMessage{requestID}:
+		return errors.New("context closed")
+	case err := <-response:
+		return err
```

Comment:
may be simpler to just guarantee that we process all inbound requests. `incomingMessages` isn't buffered anyways so we _can_ guarantee that we send back a response for all incoming messages we receive.